### PR TITLE
[Refactor] Support lazy_init for `CBGSDataset`

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

This PR is created to be consistent with `dataset_wrapper` implemented in [MMEngine](https://github.com/open-mmlab/mmengine/blob/main/mmengine/dataset/dataset_wrapper.py),

## Modification

- [x] Support lazy_init for `CBGSDataset`..
- [x] Use `force_full_init` wrapper.
- [x] Validated on `centerpoint`.
- [x] Add unit test.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by Tai-Wang** [comment]
Please add the corresponding explanation in the PR message (Motivation and Modification, etc.) for this modification.

**Comment by ZwwWayne** [comment]
This PR looks good to me, it would be great if the unit tests could be updated to cover the new usages.

**Review by JingweiZhang12 — COMMENTED** [review]
Please add UT to cover the added code.

File: tests/test_datasets/test_dataset_wrappers.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+@DATASETS.register_module()
+class CustomDataset(NuScenesDataset):
+    pass
+
+
+class TestCBGSDataset:
+
+    def setup(self):
+        dataset = NuScenesDataset
+        dataset.get_cat_ids - MagicMock(return_value=[0])
```
Comment (JingweiZhang12) [inline tests/test_datasets/test_dataset_wrappers.py:51]: This line of code looks weird

**Review by JingweiZhang12 — APPROVED** [review]

**Pull request merged**